common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/parsers/FetchContents.scala

```diff
@@ -63,10 +63,16 @@ object FetchContents {
   def write(entries: Seq[FetchEntry]): String =
     entries
       .map { e =>
-        s"${e.url} ${encodeLength(e.length)} ${encodeFilepath(e.filepath)}"
+        s"${encodeURL(e.url)} ${encodeLength(e.length)} ${encodeFilepath(e.filepath)}"
       }
       .mkString("\n")
 
+  private def encodeURL(url: URL): String =
+    url.toString.replaceAll(" ", "%20").replaceAll("\t", "%09")
```

Comment:
Is there not a generic "en / decode all URL characters" - or is there a reason we don't use it?

Reply:
Those methods exist, but the spec is a bit unclear about whether you should encode all the characters or just some of them (although I don’t think it matters).